File: Edulink/internship/models/application.py
from django.db import models
from users.models.student_profile import StudentProfile
from .internship import Internship

class Application(models.Model):
    student = models.ForeignKey(StudentProfile, on_delete=models.CASCADE, related_name='applications')
    internship = models.ForeignKey(Internship, on_delete=models.CASCADE, related_name='applications')
    application_date = models.DateTimeField(auto_now_add=True)
    status = models.CharField(
        max_length=20,
        choices=[
            ('pending', 'Pending'),
            ('reviewed', 'Reviewed'),
            ('accepted', 'Accepted'),
            ('rejected', 'Rejected'),
        ],
        default='pending'
    )
    cover_letter = models.TextField(blank=True)
    resume = models.FileField(upload_to='resumes/', blank=True)

    class Meta:
        unique_together = ('student', 'internship')

    def __str__(self):
        return f"Application by {self.student} for {self.internship.title}"

File: New folder (2)/lib/models/application.dart
enum ApplicationStatus { pending, reviewed, accepted, rejected }

class Application {
  final String id;
  final String opportunityId;
  final String studentEmail;
  final ApplicationStatus status;
  final DateTime appliedDate;
  final String? notes;

  Application({
    required this.id,
    required this.opportunityId,
    required this.studentEmail,
    required this.status,
    required this.appliedDate,
    this.notes,
  });
} 